### PR TITLE
4318: prototyp.BigInt: fix int64 parsing overflow

### PR DESCRIPTION
**Pull request by david-littlefarmer**
Ticket: https://github.com/0xsequence/issue-tracker/issues/4318

Previously, parsing a string into `BigInt` would silently return 0 on overflow of int64 without any warning. This fix ensures that numbers larger than `int64` are correctly parsed instead of returning 0, because of overflow.

Also added handling of negative numbers and tests to cover edge cases related to `int64` parsing.

### CODE REVIEW

File: lib/prototyp/bigint.go [hunk 1]
```diff
@@ -290,15 +293,34 @@ func ParseNumberString(s string, base int) (int64, bool) {
...
+func ParseNumberString(s string, base int) (int64, bool) {
+	bi, ok := ParseBigIntString(s, base)
+	if !ok {
 		return 0, false
 	}
 
-	n, err := strconv.ParseInt(s, base, 64)
-	if err != nil {
+	// Check if it fits within int64 range, if it doesn't fit in int64, return false
+	if bi.Lt(big.NewInt(math.MinInt64)) || bi.Gt(big.NewInt(math.MaxInt64)) {
```
Comment (VojtechVitek) [inline lib/prototyp/bigint.go:319]: This is perfect!

File: lib/prototyp/bigint.go [hunk 2]
```diff
@@ -290,15 +293,34 @@ func ParseNumberString(s string, base int) (int64, bool) {
...
+}
+
+func ParseNumberString(s string, base int) (int64, bool) {
+	bi, ok := ParseBigIntString(s, base)
+	if !ok {
 		return 0, false
 	}
 
-	n, err := strconv.ParseInt(s, base, 64)
-	if err != nil {
```
Comment (VojtechVitek) [inline lib/prototyp/bigint.go:297]: So this was effectively returning `0, false` for all numbers outside of the `int64` range.. which in turn caused `NewBigIntFromString()` to return `BigInt(0)` for all bigger numbers with no error.

And this is exactly [the bug](https://github.com/0xsequence/issue-tracker/issues/4318) you guys were seeing on https://market.champions.io Marketplace.

Great catch, David!

**Review by VojtechVitek — APPROVED** [review]
Looks like you fixed `NewBigIntFromString()` for all big numbers. What a great bugfix, David. Excellent work!

LGTM

**Review by VojtechVitek — APPROVED** [review]
TL;DR: This PR fixes the following bug:

```
func TestPrototyp(t *testing.T) {
	a := prototyp.NewBigIntFromString("029a2241af62c000c8", 16)
	b := prototyp.NewBigIntFromString("0x029a2241af62c000c8", 16)
	t.Logf("a: %s", a.String())
	t.Logf("b: %s", b.String())
}
```

```
=== RUN   TestPrototyp
    prototyp_test.go:12: a: 48000000000000000200
    prototyp_test.go:13: b: 0
```
